lib/cli/ui.rb

```diff
@@ -161,6 +161,26 @@ def self.raw
     ensure
       Thread.current[:no_cliui_frame_inset] = prev
     end
+
+    # Check whether colour is enabled in Formatter output. By default, colour
+    # is enabled when STDOUT is a TTY; that is, when output has not been
+    # redirected to another program or to a file.
+    #
+    def self.enable_color?
+      @enable_color
+    end
+
+    # Turn colour output in Formatter on or off.
+    #
+    # ==== Attributes
+    #
+    # * +bool+ - true or false; enable or disable colour.
+    #
+    def self.enable_color=(bool)
+      @enable_color = !!bool
+    end
+
+    self.enable_color = $stdout.tty?
```

Comment:
I would rather see this defaulted as
```ruby
def self.enable_color?
  @enable_color ||= $stdout.tty?
end
```
It seems more idiomatic and in context that way.

Reply:
Never mind.  Just realized why that's stupid: `if defined?(@enable_color)`.